src/utils/getDefaultCommitContent.js

```diff
@@ -29,8 +39,20 @@ const getDefaultCommitContent = (
     .toString()
     .split('\n')
 
+  let gitmoji = null
+  let title = commitFileContent[COMMIT_TITLE_LINE_INDEX]
+
+  const gitmojiRegexResult =
+    configurationVault.getEmojiFormat() === EMOJI_COMMIT_FORMATS.CODE
+      ? gitmojiCodeRegex.exec(title)
+      : gitmojiSymbolRegex.exec(title)
+  if (gitmojiRegexResult && gitmojiRegexResult.length === 3) {
+    ;[, gitmoji, title] = gitmojiRegexResult
+  }
```

Comment:
Overall this is fairly complex, there are some magic numbers can you explain what are you trying to do in this block?

Reply:
Basically extracting the gitmoji from the title (`3` is the size of the regex array when it succeed).

I can extract the logic to his own function to make it easier to read. Is it fair enough ?